File: CSAlgorithms/bubble/BubbleSort.cs

namespace CSAlgorithms.bubble
{
    public class BubbleSort
    {
        private int[] Arr { get; }

        public BubbleSort(int[] Arr) => this.Arr = Arr;

        public int[] Solve()
        {
            int n = Arr.Length;
            int[] output = (int[]) Arr.Clone();
            int temp;

            for (int i = 0; i < n - 1; i++)
            {
                for (int k = 0; k < n - i -1; k++)
                {
                    if (output[k] > output[k + 1])
                    {
                        temp = output[k];
                        output[k] = output[k + 1];
                        output[k + 1] = temp;
                    }
                }
            }
            return output;
        }
    }

}

File: bubble/BubbleSort.java
package me.shab.bubble;

public class BubbleSort {
	
	private final int[] arr;
	
	public BubbleSort(int arr[]) {
		this.arr = arr;
	}
	
	public int[] solve() {
		int n = this.arr.length;
		int[] output = this.arr.clone();		
		for (int i = 0; i < n-1; i++) {
			for (int j = 0; j < n-i-1; j++) {
				if (output[j] > output[j+1]) {
                    int temp = output[j];
                    output[j] = output[j+1];
                    output[j+1] = temp;
				}	
			}
		}
		return output;
	}
	
	public int[] getArray() {
		return this.arr;
	}
}
